packages/react-label/src/components/Label/Label.types.ts

```diff
@@ -46,5 +46,5 @@ export type LabelProps = Omit<ComponentProps<LabelSlots>, 'required'> &
      * an asterisk (*). Or it can be set to a string or jsx content to display a different indicator.
      * @defaultvalue false
      */
-    required?: boolean | ObjectShorthandProps<React.HTMLAttributes<HTMLElement>> | React.ReactNode;
+    required?: boolean | ComponentSlotProps<React.ComponentType<React.HTMLAttributes<HTMLElement>>> | React.ReactNode;
```

Comment:
This also should be using `IntrinsicShorthandProps`. I think this will work..?
```suggestion
    required?: boolean | IntrinsicShorthandProps<'span'>;
```